src/Resource/Collection.js

```diff
@@ -1,92 +1,119 @@
 import ParameterBag from './ParameterBag';
 
-export default class Collection {
-  /**
-   * @param {String} name
-   * @param {HyralRepository} repository
-   */
-  constructor(name, repository) {
-    /** @private */
-    this.name = name;
-
-    /** @private */
-    this.repository = repository;
+/**
+ * @param {String} name
+ * @param {HyralRepository} repository
+ */
+function Collection(name, repository) {
+  if (!new.target) {
+    throw Error('Collection() must be called with new');
+  }
 
-    /** @private */
-    this.metadata = {
-      loading: false,
-      loaded: false,
-      parameterBag: null,
-      paging: {
-        count: null,
-        pages: null,
+  Object.defineProperties(this, {
+    name: {
+      enumerable: true,
+      configurable: false,
+      writable: false,
+      value: name,
+    },
+    repository: {
+      enumerable: true,
+      configurable: false,
+      writable: false,
+      value: repository,
+    },
+    metadata: {
+      enumerable: false,
+      configurable: false,
+      value: {
+        loading: false,
+        loaded: false,
+        parameterBag: null,
+        lastParameterBagState: null,
+        paging: {
+          count: null,
+          pages: null,
+        },
       },
-    };
-
-    /** @private */
-    this.data = {
-      items: [],
-    };
-  }
+    },
+    data: {
+      enumerable: false,
+      configurable: false,
+      value: {
+        items: [],
+      },
+    },
+  });
+}
 
+Collection.prototype = {
   get parameterBag() {
     if (!this.metadata.parameterBag) {
       this.metadata.parameterBag = new ParameterBag();
     }
 
     return this.metadata.parameterBag;
-  }
+  },
 
   set parameterBag(parameterBag) {
     this.metadata.parameterBag = parameterBag;
-  }
+  },
 
   /**
    * @returns {number}
    */
   get length() {
     return this.metadata.paging.count || 0;
-  }
+  },
 
   /**
    * @returns {number}
    */
   get pages() {
     return this.metadata.paging.pages || 0;
-  }
+  },
 
   /**
    * @returns {boolean}
    */
   get isLoading() {
     return this.metadata.loading;
-  }
+  },
 
   /**
    * @returns {boolean}
    */
   get isLoaded() {
     return this.metadata.loaded;
-  }
+  },
 
   /**
    * @returns {Array}
    */
   get items() {
     return this.data.items;
-  }
+  },
 
   /**
    * @returns {Promise}
    */
   load() {
+    if (this.isLoaded === true
```

Comment:
Ik zit toch te denken of we niet iets moeten doen met als de collection aan het laden is.
Misschien niet in deze pull request oppakken. Nu is het namelijk zo dat als je heel snel na elkaar load aanroept, je toch 2 requests krijg.